Test-NetStack.psm1

```diff
@@ -865,8 +863,107 @@ Function Test-NetStack {
         }
+
+        '7' { # RDMA Stress N:1
+            if ( $ContinueOnFailure -eq $false ) {
+                if ('fail' -in $NetStackResults.Stage3.PathStatus -or 'fail' -in $NetStackResults.Stage4.PathStatus -or 'fail' -in $NetStackResults.Stage5.ReceiverStatus -or 'fail' -in $NetStackResults.Stage6.NetworkStatus) {
+    
+                    $Stage -ge 7 | ForEach-Object {
+                        $AbortedStage = $_
+                        $NetStackResults | Add-Member -MemberType NoteProperty -Name "Stage$AbortedStage" -Value 'Aborted'; $StageFailures++
+                    }
+    
+                    Write-Warning 'Aborted due to failures in earlier stage(s). To continue despite failures, use the ContinueOnFailure parameter.'
+                    return $NetStackResults
+                }
+            }
+            $NodeGroups = $Mapping | Where-Object VLAN -ne 'Unsupported' | Group-Object NodeName
+            $thisStage = $_
+            Write-Host "Beginning Stage: $thisStage - RDMA Perf VMSwitch Stress - $([System.DateTime]::Now)"
+            "Stage 7`r`n" | Out-File $LogFile -Append -Encoding utf8 -Width 2000
+            "Console Output" | Out-File $LogFile -Append -Encoding utf8 -Width 2000
+            "Beginning Stage: $thisStage - RDMA Perf VMSwitch Stress - $([System.DateTime]::Now)" | Out-File $LogFile -Append -Encoding utf8 -Width 2000
+            $StageResults = @()
+            $ISS = [System.Management.Automation.Runspaces.InitialSessionState]::CreateDefault()
+            $NetStackHelperModules = Get-ChildItem (Join-Path -Path $PSScriptRoot -ChildPath 'Helpers\*') -Include '*.psm1'
+            $NetStackHelperModules | ForEach-Object { $ISS.ImportPSModule($_.FullName) }
+            $Max = [int]$env:NUMBER_OF_PROCESSORS * 2
+            $NodeGroups | ForEach-Object {
+                $GroupedJobs = @()            
+                $RunspacePool = [runspacefactory]::CreateRunspacePool(1, $Max, $ISS, $host)
+                $RunspacePool.Open()
+                $testNodeGroup = $_  
+                $testNodeGroup.Group | Where-Object -FilterScript { $_.RDMAEnabled } | ForEach-Object {
+                    
+                    $thisSource  = $_
+                    $PowerShell = [powershell]::Create()
+                    $PowerShell.RunspacePool = $RunspacePool
+
+                    $ClientNodes = @($Mapping | Where-Object NodeName -ne $thisSource.NodeName | Where-Object VLAN -eq $thisSource.VLAN | Where-Object Subnet -eq $thisSource.Subnet | Where-Object -FilterScript { $_.RDMAEnabled })
+                    [void] $PowerShell.AddScript({
+                        param ( $thisSource, $ClientNodes, $Definitions )
+
+                        try {
+                            $thisSourceResult = Invoke-NDKPerfNto1 -Server $thisSource -ClientNetwork $ClientNodes -ExpectedTPUT $Definitions.NDKPerf.TPUT
+                        } catch { throw $_.exception.message }
```

Comment:
I haven't tried this, but exceptions don't tend to be overly friendly messages...Is this a user friendly/appropriate message?

Reply:
Accidental leftover from debugging, I'll remove it.